src/main/java/org/cactoos/scalar/NumberEnvelope.java

```diff
@@ -41,78 +41,34 @@
      */
     private static final long serialVersionUID = -1924406337256921883L;
 
-    /**
-     * The LONG number.
-     */
-    private final Scalar<Long> lnum;
-
-    /**
-     * The INT number.
-     */
-    private final Scalar<Integer> inum;
+    private final Number origin;
 
-    /**
-     * The FLOAT number.
-     */
-    private final Scalar<Float> fnum;
-
-    /**
-     * The DOUBLE number.
-     */
-    private final Scalar<Double> dnum;
-
-    /**
-     * Ctor.
-     * @param dnm Double scalar
-     */
-    public NumberEnvelope(final Scalar<Double> dnm) {
-        this(
-            () -> dnm.value().longValue(),
-            () -> dnm.value().intValue(),
-            () -> dnm.value().floatValue(),
-            dnm
-        );
-    }
-
-    /**
-     * Ctor.
-     * @param lnm Long scalar
-     * @param inm Integer scalar
-     * @param fnm Float scalar
-     * @param dnm Long scalar
-     * @checkstyle ParameterNumberCheck (5 lines)
-     */
-    public NumberEnvelope(final Scalar<Long> lnm, final Scalar<Integer> inm,
-        final Scalar<Float> fnm, final Scalar<Double> dnm) {
-        super();
-        this.lnum = lnm;
-        this.inum = inm;
-        this.fnum = fnm;
-        this.dnum = dnm;
+    public NumberEnvelope(final Number number) {
+        this.origin = number;
     }
 
     @Override
-    public final int intValue() {
-        return new Unchecked<>(this.inum).value();
+    public Double value() throws Exception {
```

Comment:
@ruca905 please remove the implementation of `Scalar<Double>`, it's out of scope of `NumberEnvelope`.